lib/core/device.rb

```diff
@@ -835,7 +835,12 @@ def get_contexts(action)
 
   # parses and saves the source code for currect page.
   def get_source(action)
-    source = @driver.get_source
+    source = nil
+    begin
+      source = @driver.get_source
+    rescue => e 
```

Comment:
surely we can branch depending on whether `app = desktop`, instead of a begin/rescue?